src/chromo/kinematics.py

```diff
@@ -230,11 +236,44 @@ def __init__(
 
         _gamma_cm = (elab + m2) / ecm
         _betagamma_cm = plab / ecm
+        self.m1 = m1
+        self.m2 = m2
 
         super().__init__(
             frame, part1, part2, ecm, plab, elab, ekin, beams, _gamma_cm, _betagamma_cm
         )
 
+        self._beam_data = None
+
+    def _get_beam_data(self, generator_frame):
```

Comment:
`_beam_data`  is not a constant of `kinematics` object set once and for all. It depends also on event generator (`generator_frame`) that uses that the `kinematics` object. It could not be a property, i.e. getter without argument.